Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/ContainerActionButtonHandler.cpp

```diff
@@ -42,4 +44,30 @@ namespace AzToolsFramework
             break;
         }
     }
+
+    void ContainerActionButtonHandler::OnClicked()
+    {
+        using AZ::DocumentPropertyEditor::Nodes::Container;
+        using AZ::DocumentPropertyEditor::Nodes::ContainerAction;
+
+        if (m_action == ContainerAction::Clear)
+        {
+            // Default to true if the PromptOnContainerClear attribute isn't present
+            bool shouldPromptOnClear = Container::PromptOnContainerClear.ExtractFromDomNode(m_node).value_or(true);
+
+            if (shouldPromptOnClear)
+            {
+                auto result = QMessageBox::question(
```

Comment:
don't we have a styled version of this that we're supposed to use now? I vaguely recall LY having a QMessageBox replacement... ignore me, if you don't know what I'm talking about...

Reply:
We don't have AZ versions of the `QMessageBox`, because we already override their style in the StyleManager, so we get the styled version for free without having to change all the API calls. You might be thinking of the `AzMessageBox` variants we have, but they aren't a separate class they just let you specify a setreg first so that you can get the behavior of "Do not show this message again" checkbox functionality.